src/Elliptic/DiscontinuousGalerkin/Actions/ApplyOperator.hpp

```diff
@@ -0,0 +1,629 @@
+    tmpl::list<FluxesArgsTags...>, tmpl::list<SourcesArgsTags...>> {
+ private:
+  static constexpr size_t Dim = System::volume_dim;
+  using all_mortar_data_tag = ::Tags::Mortars<
+      elliptic::dg::Tags::MortarData<typename TemporalIdTag::type,
+                                     typename PrimalMortarFieldsTag::tags_list,
+                                     typename PrimalMortarFluxesTag::tags_list>,
+      Dim>;
+  using mortar_data_inbox_tag =
+      MortarDataInboxTag<Dim, TemporalIdTag,
+                         typename PrimalMortarFieldsTag::tags_list,
+                         typename PrimalMortarFluxesTag::tags_list>;
+
+ public:
+  using const_global_cache_tags =
+      tmpl::list<elliptic::dg::Tags::PenaltyParameter>;
+  using inbox_tags = tmpl::list<mortar_data_inbox_tag>;
+
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex>
+  static bool is_ready(const db::DataBox<DbTags>& box,
+                       const tuples::TaggedTuple<InboxTags...>& inboxes,
+                       const Parallel::GlobalCache<Metavariables>& /*cache*/,
+                       const ArrayIndex& /*array_index*/) noexcept {
+    return ::dg::has_received_from_all_mortars<mortar_data_inbox_tag>(
+        db::get<TemporalIdTag>(box), get<domain::Tags::Element<Dim>>(box),
+        inboxes);
+  }
+
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTags>&&> apply(
+      db::DataBox<DbTags>& box, tuples::TaggedTuple<InboxTags...>& inboxes,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    const auto& temporal_id = get<TemporalIdTag>(box);
+
+    // Move received "remote" mortar data into the DataBox
+    if (LIKELY(db::get<domain::Tags::Element<Dim>>(box).number_of_neighbors() >
+               0)) {
+      auto received_mortar_data =
+          std::move(tuples::get<mortar_data_inbox_tag>(inboxes)
+                        .extract(temporal_id)
+                        .mapped());
+      db::mutate<all_mortar_data_tag>(
+          make_not_null(&box), [&received_mortar_data, &temporal_id](
+                                   const auto all_mortar_data) noexcept {
+            for (auto& [mortar_id, mortar_data] : received_mortar_data) {
+              all_mortar_data->at(mortar_id).remote_insert(
+                  temporal_id, std::move(mortar_data));
+            }
+          });
+    }
+
+    // Apply DG operator
+    db::mutate<OperatorAppliedToFieldsTag, all_mortar_data_tag>(
+        make_not_null(&box),
+        [](const auto operator_applied_to_fields, const auto all_mortar_data,
+           const auto& primal_fields, const auto& primal_fluxes,
+           const auto&... args) noexcept {
```

Comment:
Pushed a fixup that forwards all const-ref args. Is there a way to get rid of the lambda entirely, or forward all arguments including the not-null pointers? Just passing `&elliptic::dg::apply_operator<System, Linearized>` instead of the lambda doesn't work because the function infers its template parameters from arguments.

Reply:
Of course 🤦‍♂️ the pointers can also just be forwarded as const-refs..